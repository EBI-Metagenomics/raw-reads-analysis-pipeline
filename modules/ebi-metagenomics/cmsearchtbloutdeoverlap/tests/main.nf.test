nextflow_process {

    name "Test Process CMSEARCHTBLOUTDEOVERLAP"
    script "../main.nf"
    process "CMSEARCHTBLOUTDEOVERLAP"

    tag "modules"
    tag "modules_nfcore"
    tag "cmsearchtbloutdeoverlap"

    test("cmsearchtbloutdeoverlap") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],                                       // meta map
                    file("${moduleDir}/tests/1.cmscan.clan.tblout", checkIfExists: true)   // cmsearch out
                    ]
                input[1] = [file("${moduleDir}/tests/ribo.claninfo", checkIfExists: true)] // claninfo
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
                

    }

    test("cmsearchtbloutdeoverlap - decompress") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],                                          // meta map
                    file("${moduleDir}/tests/1.cmscan.clan.tblout.gz", checkIfExists: true)   // cmsearch out
                    ]
                input[1] = [file("${moduleDir}/tests/ribo.claninfo", checkIfExists: true)]    // claninfo
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
                

    }

    test("cmsearchtbloutdeoverlap - primer validation") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],                                          // meta map
                    file("${moduleDir}/tests/f_and_r_primers.tblout", checkIfExists: true)   // cmsearch out
                    ]
                input[1] = [file("${moduleDir}/tests/ribo.claninfo", checkIfExists: true)]    // claninfo
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
                

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],                                       // meta map
                    file("${moduleDir}/tests/1.cmscan.clan.tblout", checkIfExists: true)   // cmsearch out
                    ]
                input[1] = [file("${moduleDir}/tests/ribo.claninfo", checkIfExists: true)] // claninfo
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
