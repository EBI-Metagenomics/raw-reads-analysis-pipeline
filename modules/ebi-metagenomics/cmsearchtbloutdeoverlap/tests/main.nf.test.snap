{
    "cmsearchtbloutdeoverlap - decompress": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,c68f9407a07e6b056025ca97001b5ef8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ],
                "cmsearch_tblout_deoverlapped": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,c68f9407a07e6b056025ca97001b5ef8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-05-09T13:36:16.134340556"
    },
    "cmsearchtbloutdeoverlap - primer validation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,9cd7612a1fbf53ccbe29c8b8875e3559"
                    ]
                ],
                "1": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ],
                "cmsearch_tblout_deoverlapped": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,9cd7612a1fbf53ccbe29c8b8875e3559"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-05-09T13:40:13.316000216"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ],
                "cmsearch_tblout_deoverlapped": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-05-09T13:36:25.527946847"
    },
    "cmsearchtbloutdeoverlap": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,c68f9407a07e6b056025ca97001b5ef8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ],
                "cmsearch_tblout_deoverlapped": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tblout.deoverlapped:md5,c68f9407a07e6b056025ca97001b5ef8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,86775aca65189c456e0982d1bbd95ace"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-05-09T13:35:53.001130738"
    }
}